api/v1beta1/bgppeer_types.go

```diff
@@ -81,6 +82,10 @@ type BGPPeerSpec struct {
 	// +optional
 	Password string `json:"password,omitempty" yaml:"password,omitempty"`
 
+	// secretRef is name of the authentication secret for BGP Peer
+	// +optional
+	SecretRef *v1.LocalObjectReference `json:"secretRef,omitempty" yaml:"secretRef,omitempty"`
```

Comment:
I'd rename this as PasswordSecret, meaning the secret that holds the password.

Reply:
done